lib/doorkeeper/models/concerns/hashable.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Doorkeeper
+  module Models
+    ##
+    # Hashable finder to provide lookups for input plaintext values which are
+    # mapped to their hashes before lookup.
+    module Hashable
```

Comment:
I prefer `Encryptable` over `Hashable`. However, RFC 6819 allows both - encrypt / hash credentials (check out 5.1.4.1.4.  Encryption of Credentials for example).

BTW, very good point to extract this functionality to separate concern that could be reused in any Doorkeeper gem extension 👍

Reply:
Again, as above, the option added is `one-way` and thus will not allow for encryption of token values. It's possible to alter that, but with my rationale in the opening statement, we should avoid encryption in all cases where authorization flow does not require it. Since doorkeeper implements no such scheme where e.g., the `client_secret` needs to be accessed in clear text, there is no advantage of encryption over hashing.

Reply:
Yep, now I understand. Agree